pkg/router/template/util/util.go

```diff
@@ -0,0 +1,71 @@
+package util
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/golang/glog"
+
+	routeapi "github.com/openshift/origin/pkg/route/apis/route"
+)
+
+// Generate a regular expression to match route hosts (and paths if any).
+func GenerateRouteRegexp(hostname, path string, wildcard bool) string {
+	hostRE := regexp.QuoteMeta(hostname)
+	if wildcard {
+		subdomain := routeapi.GetDomainForHost(hostname)
+		if len(subdomain) == 0 {
+			glog.Warningf("Generating subdomain wildcard regexp - invalid host name %s", hostname)
+		} else {
+			subdomainRE := regexp.QuoteMeta(fmt.Sprintf(".%s", subdomain))
+			hostRE = fmt.Sprintf(`[^\.]*%s`, subdomainRE)
+		}
+	}
+
+	portRE := "(:[0-9]+)?"
+
+	// build the correct subpath regex, depending on whether path ends with a segment separator
+	var pathRE, subpathRE string
+	switch {
+	case strings.TrimRight(path, "/") == "":
```

Comment:
i think we use len() == 0 convention to check empty strings

Reply:
ok - will do (its existing code that got moved to a new file).